paz/backend/standard.py

```diff
@@ -271,3 +271,26 @@ def predict(x, model, preprocess=None, postprocess=None):
     if postprocess is not None:
         y = postprocess(y)
     return y
+
+
+def predict_batch(x, model, preprocess=None, postprocess=None):
+    """Preprocess, predict and postprocess batched input.
+    # Arguments
+        x: Input to model
+        model: Callable i.e. Keras model.
+        preprocess: Callable, used for preprocessing input x.
+        postprocess: Callable, used for postprocessing output of model.
+
+    # Note
+        If model outputs a tf.Tensor is converted automatically to numpy array.
+    """
+    if preprocess is not None:
+        x = preprocess(x)
+    if x is None:
+        return None
```

Comment:
I am not sure that this function should be called `predict_batch`. The only difference is the check `if x is None`. Maybe something like `predict_if_not_none`. I know is not clean enough, but both functions are so close to each other that we don't know really if this is the way all batched models will predict.

Reply:
Wurde zu predict_noneable umgeändert